<launch>

	<node pkg="viso2_ros" type="stereo_odometer" name="stereo_odometer" output="screen">
		<!-- Main() -->
		<param name="visualisation_on" type="bool" value="true" /> 

		<!-- Stereo Processor -->
		<param name="left_topic" type="str" value="/stereo/left/image_rect_raw" />
		<param name="right_topic" type="str" value="/stereo/right/image_rect_raw" />
		<param name="left_info_topic" type="str" value="/stereo/left/camera_info" />
		<param name="right_info_topic" type="str" value="/stereo/right/camera_info" />
		<param name="queue_size" type="int" value="3" />

		<!-- QuadMatcher -->
		<param name="number_of_buckets_in_width" type="int" value="8" />
		<param name="number_of_buckets_in_height" type="int" value="8" />
		<param name="epipolar_tolerance" type="double" value="10" />
		<param name="max_neighbor_ratio" type="double" value="0.7" />
		<param name="use_bucketing" type="bool" value="true" /> 
		<param name="compulte_scaled_keys" type="bool" value="true" /> 

		<!-- StereoMotionEstimator -->
		<param name="ransac_iters" type="int" value="400" />
		<param name="inlier_threshold" type="double" value="10.0" />
		<param name="inlier_ratio_min" type="double" value="0.5" />
		<param name="reweighting" type="bool" value="true" />

		<!-- Stereo Odometry Main Class -->
		<param name="ref_frame_change_method" type="str" value="flow" /> <!-- "flow" "inlier" -->
		<param name="ref_frame_motion_threshold" type="double" value="30.0" />
		<param name="ref_frame_inlier_threshold" type="int" value="12" />

		<param name="image_pre_scaling" type="double" value="1" />
		<param name="image_pre_contrast" type="double" value="1.6" />   
		<param name="image_pre_brightness" type="double" value="60.0" />

		<param name="noise_translation" type="double" value="1.0e-3" />   
		<param name="noise_rotation" type="double" value="1.0e-4" /> 

		<param name="change_delta_pose_to_velocity" type="bool" value="true" /> 
		<!-- <param name="moving_object_polygons" type="str" value="/arena_movidius/objects" /> -->
	</node>

</launch>
